src/trcb.erl

```diff
@@ -36,37 +36,41 @@
 
 %% @doc check if a message should be deliver and deliver it, if not add it to the queue
 -spec causal_delivery({actor(), message(), timestamp()}, timestamp(), [{actor(), message(), timestamp()}], fun()) -> {timestamp(), [{actor(), message(), timestamp()}]}.
-causal_delivery({Origin, Msg, MsgVV}, VV, Queue, Function) ->
-    case vclock:dominates(MsgVV, VV) of
+causal_delivery({Origin, MessageBody, MessageVClock}, VV, Queue, Function) ->
+    lager:info("Our Clock: ~p", [VV]),
+    lager:info("Incoming Clock: ~p", [MessageVClock]),
+    case vclock:dominates(MessageVClock, VV) of
         true ->
+            %% TODO: Why is this increment operation here?
             NewVV = vclock:increment(Origin, VV),
```

Comment:
Ỳou're right. This increment should be above the line
`try_to_deliever(Queue, {NewVV, Queue}, Function)`